shells/backends/gcp_tf_backend/src/driver.py

```diff
@@ -59,64 +58,91 @@ def get_inventory(self, context):
         try:
             storage_client = storage.Client()
             get_bucket = storage_client.get_bucket(bucket_name)
             if len(str(get_bucket)) < 0:
-                raise ValueError(f"Bucket {bucket_name} not found")  
+                raise ValueError(f"Bucket {bucket_name} not found")
         except Exception as e:
-            self._raise_and_log(logger, f"There was an issue accessing the bucket {bucket_name}.{e}")        
+            self._raise_and_log(logger, f"There was an issue accessing the bucket {bucket_name}.{e}")
 
-    def _can_conntect_to_gcp(self, context, logger):
+    def _can_conntect_to_gcp(self, context, logger) -> bool:
         gcp_backend_resource = GcpTfBackend.create_from_context(context)
         project_id = gcp_backend_resource.project
-        create_session=self._create_gcp_session(context, project_id, logger)
+        self._create_gcp_session(context, project_id, logger)
         client = discovery.build('compute', 'v1')
         response = client.healthChecks().list(project=project_id).execute()
         return len(response) > 0
 
-    def _generate_state_file_string(self, gcp_backend_resource: GcpTfBackend):
+    def get_backend_data(self, context, tf_state_unique_name: str) -> str:
+        self._backend_secret_vars = {}
+
+        with LoggingSessionContext(context) as logger:
+            gcp_backend_resource = GcpTfBackend.create_from_context(context)
+            tf_state_file_string = self._generate_state_file_string(gcp_backend_resource, tf_state_unique_name)
+            backend_data = {"tf_state_file_string": tf_state_file_string}
+            project_id = gcp_backend_resource.project
+            try:
+                gcp_service = self._create_gcp_session(context, project_id, logger)
+                self._backend_secret_vars = {"credentials": gcp_service}
+            except Exception as e:
+                self._raise_and_log(logger, "Inputs for Cloud Backend Access missing or incorrect")
+
+            logger.info(f"Returning backend data for creating provider file :\n{backend_data}")
+            response = json.dumps({"backend_data": backend_data, "backend_secret_vars": self._backend_secret_vars})
+            return response
+
+    def delete_tfstate_file(self, context, tf_state_unique_name: str):
+        """Deletes a blob from the bucket."""
+        with LoggingSessionContext(context) as logger:
+            gcp_backend_resource = GcpTfBackend.create_from_context(context)
+            project_id = gcp_backend_resource.project
+            try:
+                gcp_service = self._create_gcp_session(context, project_id, logger)
+                storage_client = storage.Client()
+                bucket_name = gcp_backend_resource.bucket_name
+                bucket = storage_client.get_bucket(bucket_name)
+                """Delete object under folder"""
+                blobs = list(bucket.list_blobs(prefix=tf_state_unique_name))
```

Comment:
there should always be just one tf state file. Because we are using "prefix" and not full name match there is a possibility that we will delete additional files that shouldn't be deleted.
If there is an option to use exact name match it will preferred.
If not, then I suggest to add a validation to make sure that we match only one blob.

Reply:
so, this tf_state_unique_name is a folder name, and it will delete only such folder, in our case prefix=folder name, which is very long, and belongs to name of sandbox. For example name of the folder(which is prefix here) oleksandr-r-test-tf/4f9b32f6-6407-4fe4-870e-f3e4b6f9d883_19c611a952be41849df425b418ed92a7.tf.state and file name inside it  default.tfstate